File: src/main/java/it/contrader/automative/model/Dispositivo.java
package it.contrader.automative.model;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

@Entity
@Table(name = "dispositivo")
public class Dispositivo implements Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int id;
	
	@Column
	private String codice;
	
	@OneToOne(fetch=FetchType.LAZY)
	@JoinColumn(name = "idAuto")
	private Auto auto;
	
	@Column
	private Date dataInstallazione;
	
	@OneToOne(fetch=FetchType.LAZY)
	@JoinColumn(name = "idOfficina")
	private Officina officina;
}

File: FrontAutomotiveAngular5/src/app/models/Dispositivo.ts
import { Officina } from './Officina';
import { Auto } from './Auto'; 

export class Dispositivo {
    private id: number;
    private codice: string;
    private auto: Auto;
    private dataInstallazione: string;
    private officina: Officina;
   


	constructor(id: number,codice:string,auto:Auto,dataInstallazione:string,officina:Officina) {
		this.id = id;
		this.codice = codice;
		this.auto = auto;
		this.dataInstallazione = dataInstallazione;
		this.officina = officina;
		
	}
        
}